### DISCUSSION TITLE
Audio filters? (Underwater sound etc)

### DISCUSSION THREAD

**Discussion by reflekshun** [HaxeFlixel/flixel#2388] [Q&A]
Is it at all possible to use audio filters with HaxeFlixel? I have underwater sections in a game I'm working on, and I would love to be able to make the music sound more underwater during those sections. I know it's a long shot but thought I'd ask in case anyone knew a smart way around it to make that work. Thanks!

**Comment by RichardBray** [comment]
So I haven't used this myself but if you're okay with middleware I think [FMOD](https://www.fmod.com/) might be able to do something like that. There is a [Haxe plugin](https://lib.haxe.org/p/haxefmod/) for it which means you should be able to use it in your game. 

Otherwise just separate soundtracks for underwater and not underwater I guess 🤷

**Reply by reflekshun** [reply]
Thanks for these suggestions! I'm not sure if the dev (not me, I'm making the music) wants middleware, but it's a last ditch option if there is a free option somehow for FMOD or wwise. Not sure if there is since we will be selling it on steam so technically it's a commercial product, just a very small one! 

Love the idea on separate soundtracks and will definitely give that a shot. Is there a way to make an event trigger volume changes? My idea is to have both filtered and non-filtered tracks running, then swap out volumes of each track depending on whether you have crossed into water, or crossed out of water, if that makes sense. That way there will be a smoother crossfade without any clicks or glitches (in theory).

Thanks again so much for the help!

**Reply by reflekshun** [reply]
Just writing back to say no need to stress on the middleware, looks like the dev figured it out! Works great, so to anyone who has the desire to have this effect, it's totally possible in Haxeflixel! Haxeflixel does a good job of preserving timestamp on a music track, so you can quite smoothly swap between music tracks without clicks, at least that's our experience so far 👍
